### ISSUE TITLE
Deleting Party Relation

### ISSUE THREAD

**Issue by bcooney93** [karbonfm/karbonfm#88]
Deleting a party relation deletes more than the two representative records.

**Comment by bcooney93** [comment]
Currently, we delete a contact from an Opportunity by deleting the PartyRelationship record. This isn't correct, as there are two PartyRelationship records. Perhaps we should instead, use EndRelationship?

**Comment by bcooney93** [comment]
Also, we need to improve the UI to more clearly indicate a relationship that has ended.

**Comment by davegraham92120** [comment]
The original issue that deleting one PartyRelation was not just deleting the reverse relationship record, but also was deleting additional records not associated with the relationship.

I reopened this so I can get clarity on the discussion comments, because I think multiple issues might have been addressed in this bug report: 

1. We currently don't have an OpportunityParty entity, so there shouldn't be a way to remove a party from an Opportunity, as it would be confusing to the user. If we decide to support that feature, it would require its own feature request.
2. In the interest of preserving the integrity of temporal data, ending a relationship is preferred over deleting the records. That said, this would be a new feature request since there are many factors to consider (e.g., updating/adding API, updating UI to show active vs. historical).
